ui/app/components/exec/index.ts

```diff
@@ -0,0 +1,54 @@
+import Component from '@glimmer/component';
+import { Empty } from 'google-protobuf/google/protobuf/empty_pb';
+import { ExecStreamRequest } from 'waypoint-pb';
+import { ExecWebSocketAddon } from 'waypoint/utils/exec-websocket-addon';
+import SessionService from 'ember-simple-auth/services/session';
+import { Terminal } from 'xterm';
+import { createTerminal } from 'waypoint/utils/create-terminal';
+import { inject as service } from '@ember/service';
+import { tracked } from '@glimmer/tracking';
+
+interface ExecComponentArgs {
+  deploymentId: string;
+}
+
+export default class ExecComponent extends Component<ExecComponentArgs> {
+  @service session!: SessionService;
+
+  @tracked deploymentId: string;
+  @tracked terminal!: Terminal;
+  @tracked socket!: WebSocket;
+
+  constructor(owner: unknown, args: ExecComponentArgs) {
+    super(owner, args);
+    let { deploymentId } = this.args;
+    this.deploymentId = deploymentId;
+    this.terminal = createTerminal({ inputDisabled: false });
+    this.terminal.focus();
+    this.startExecStream(deploymentId);
+  }
+
+  async startExecStream(deploymentId: string): Promise<void> {
+    let token = this.session.data.authenticated?.token;
+    let apiHost = window.location.hostname;
+    let socket = new WebSocket(`wss://${apiHost}:9702/v1/exec?token=${token}`);
```

Comment:
**Note to self:** I wonder if we can determine the port dynamically here? Just trying this out against an EKS installation of Waypoint where I’ve mapped 9702 on the server to 80 on the ingress.

Reply:
Addressed in 7c8572b96877cb9b56fa9e5a8b22201e8291ec48

Reply:
Didn’t quite cover all my bases in that commit. Rectified in https://github.com/hashicorp/waypoint/pull/2856